internal/service/common/device.go

```diff
@@ -22,20 +22,45 @@ const (
 	DeviceStatusInfoRebooting      = "The device is rebooting."
 )
 
-func UpdateServiceSideStatus(ctx context.Context, st store.Store, log logrus.FieldLogger, orgId uuid.UUID, device *api.Device) bool {
+func UpdateServiceSideStatus(ctx context.Context, st store.Store, log logrus.FieldLogger, orgId uuid.UUID, device, oldDevice *api.Device) []api.ResourceUpdatedDetails {
 	if device == nil {
-		return false
+		return nil
 	}
 	if device.Status == nil {
 		status := api.NewDeviceStatus()
 		device.Status = &status
 	}
 
-	deviceStatusChanged := updateServerSideDeviceStatus(device)
-	updatedStatusChanged := updateServerSideDeviceUpdatedStatus(ctx, st, log, orgId, device)
-	applicationStatusChanged := updateServerSideApplicationStatus(device)
-	lifecycleStatusChanged := updateServerSideLifecycleStatus(device)
-	return deviceStatusChanged || updatedStatusChanged || applicationStatusChanged || lifecycleStatusChanged
+	updatedDetails := []api.ResourceUpdatedDetails{}
```

Comment:
We said that some of these would be different events.  None of them would be device-updated events.